android/PartialSync/app/src/main/java/io/realm/todo/WelcomeActivity.java

```diff
@@ -0,0 +1,123 @@
+    private View mProgressView;
+    private View mLoginFormView;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_welcome);
+
+        if (SyncUser.currentUser() != null) {
+            setUpRealmAndGoToListTaskActivity();
+        }
+
+        // Set up the login form.
+        mNicknameTextView = findViewById(R.id.nickname);
+        Button loginButton = findViewById(R.id.login_button);
+        loginButton.setOnClickListener(view -> attemptLogin());
+        mLoginFormView = findViewById(R.id.login_form);
+        mProgressView = findViewById(R.id.login_progress);
+    }
+
+    private void attemptLogin() {
+        // Reset errors.
+        mNicknameTextView.setError(null);
+        // Store values at the time of the login attempt.
+        String nickname = mNicknameTextView.getText().toString();
+        showProgress(true);
+
+        SyncCredentials credentials = SyncCredentials.nickname(nickname, true);
+        SyncUser.loginAsync(credentials, AUTH_URL, new SyncUser.Callback<SyncUser>() {
+            @Override
+            public void onSuccess(SyncUser user) {
+                showProgress(false);
+                setUpRealmAndGoToListTaskActivity();
+            }
+
+            @Override
+            public void onError(ObjectServerError error) {
+                showProgress(false);
+                mNicknameTextView.setError("Uh oh something went wrong! (check your logcat please)");
+                mNicknameTextView.requestFocus();
+                Log.e("Login error", error.toString());
+            }
+        });
+    }
+
+    /**
+     * Shows the progress UI and hides the login form.
+     */
+    @TargetApi(Build.VERSION_CODES.HONEYCOMB_MR2)
+    private void showProgress(final boolean show) {
+        int shortAnimTime = getResources().getInteger(android.R.integer.config_shortAnimTime);
+        mLoginFormView.setVisibility(show ? View.GONE : View.VISIBLE);
+        mLoginFormView.animate().setDuration(shortAnimTime).alpha(
+                show ? 0 : 1).setListener(new AnimatorListenerAdapter() {
+            @Override
+            public void onAnimationEnd(Animator animation) {
+                mLoginFormView.setVisibility(show ? View.GONE : View.VISIBLE);
+            }
+        });
+        mProgressView.setVisibility(show ? View.VISIBLE : View.GONE);
+        mProgressView.animate().setDuration(shortAnimTime).alpha(
+                show ? 1 : 0).setListener(new AnimatorListenerAdapter() {
+            @Override
+            public void onAnimationEnd(Animator animation) {
+                mProgressView.setVisibility(show ? View.VISIBLE : View.GONE);
+            }
+        });
+    }
+
+    private void setUpRealmAndGoToListTaskActivity(){
+        SyncConfiguration configuration = new SyncConfiguration.Builder(
+                SyncUser.currentUser(),
+                REALM_URL + "/items")
```

Comment:
Maybe make the URL in `Constants` also include `items`? It is a little confusing to have a parameter called `REALM_URL` and then it is really `REALM_BASE_URL`